National Gallery of Australia.js

```diff
@@ -0,0 +1,98 @@
+{
+	"translatorID": "ac9c95f6-7692-47fb-b231-4cc36d1d67f5",
+	"label": "National Gallery of Australia",
+	"creator": "Ben Swift",
+	"target": "^https?://searchthecollection\\.nga\\.gov\\.au/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-08-18 02:43:18"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2021 Ben Swift
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes("uniqueId=")) {
+		return "artwork";
+	}
+
+	return false;
+}
+
+function parseArtwork(doc, url){
+
+	var selectorMap = {
+		artist: ".main-artist-name > span:nth-child(1) > a:nth-child(1)",
```

Comment:
Try to avoid `:nth-child` selectors when possible. They're almost certain to break quickly or already be broken on pages that you haven't tested on. I see that the site doesn't have classes that can easily be matched on, but it _is_ populated via JSON. Why not load the data the same way the site does? Make a GET request:

```js
let uniqueID = ...; // extract from the URL
let jsonURL = `https://searchthecollection.nga.gov.au/stcapi/service/stc/node?uniqueId=${uniqueID}`;
ZU.doGet(jsonURL, function (respText) {
	let json = JSON.parse(respText);

	let title = json.vraTitleDisplayText;

	// basic creator info:
	let creators = json.creators; // ["Alison Alder", "Redback Graphix", "NSW Women's Advisory Service", "Printcraft Pty Ltd"]

	// or, maybe better:
	let creators = JSON.parse(json.creatorsJson); // an array of objects, as follows:
	for (let creator of creators) {
		let name = creator.NamFullName; // "Alison Alder"
		let type = creator.NamPartyType; // "Person" or "Organisation"
		if (type == "Person") {
			// if we get a person, we should treat their name like
			// a personal name and attempt to parse it
			item.creators.push(ZU.cleanAuthor(name, 'artist'));
		}
		else {
			// if we get an organization, we should pass its name
			// through as-is
			item.creators.push({
				lastName: name,
				creatorType: 'contributor', // organizations probably aren't artists?
				fieldMode: 1
			});
		}
	}

	// do more processing...
});
```

This way we can avoid doing messy, fragile screen scraping altogether. I hope this helps!

Reply:
(Examples using https://searchthecollection.nga.gov.au/object?uniqueId=4193).

Another advantage of using the JSON API is that we can handle multiples - don't even need to load the front-facing HTML for each one, just pull the `uniqueId` from each link and grab the JSON.